code/jasmin/mlkem_ref/kem.jinc

```diff
@@ -25,31 +25,25 @@ fn __crypto_kem_keypair_jazz(reg u64 pkp, reg u64 skp, reg ptr u8[MLKEM_SYMBYTES
 
   for i=0 to MLKEM_INDCPA_PUBLICKEYBYTES/8
   {
-    t64 = (u64)[pkp + 8*i];
-    (u64)[skp] = t64;
-    skp += 8;
+    (u64)[skp + 8 * i] = (u64)[pkp + 8 * i];
   }
 
-  s_skp = skp;
+  s_skp += MLKEM_POLYVECBYTES + MLKEM_INDCPA_PUBLICKEYBYTES;
   pkp = s_pkp;
   t64 = MLKEM_POLYVECBYTES + MLKEM_SYMBYTES;
   h_pk = _isha3_256(h_pk, pkp, t64);
   skp = s_skp;
 
   for i=0 to 4
   {
-    t64 = h_pk[u64 i];
-    (u64)[skp] = t64;
-    skp += 8;
+    (u64)[skp + 8 * i] = h_pk[u64 i];
```

Comment:
Will memory-to-memory assignments be supported in the compiler in the long run? (the compiler inserts the extra assignment that is needed).
Is this already the case for `+=` and similar?

I recall that a while ago, I discussed with Benjamin that it might not be the case (but it was a long time ago, so things might have changed).

In any case, I agree that these statements look better like this (but the corresponding compiler warning is annoying).

Reply:
Is there a warning? There should not be.

Reply:
I see that there is no warning. It seems that the compiler is inserting extra instrucions without reporting: is this the case?